api/Infrastructure/Authentication/AuthenticationServiceCollectionExtension.cs

```diff
@@ -254,102 +254,155 @@ private static async Task OnPostAuthSuccess(
         {
             var judgeId = configuration.GetNonEmptyValue("PCSS:JudgeId");
             var homeLocationId = configuration.GetNonEmptyValue("PCSS:JudgeHomeLocationId");
-
-            // Default the user to his own external Judge Id.
-            if (!string.IsNullOrWhiteSpace(context.Principal.ExternalJudgeId()))
+            PersonSearchItem judge = null;
```

Comment:
refactored this large method into smaller pieces, but this could be further simplified to allow for easier mocking in unit tests.